networkx/algorithms/structuralholes.py

```diff
@@ -144,16 +152,18 @@ def redundancy(G, u, v, weight=None):
     # Use Borgatti's simplified formula for unweighted and undirected graphs
     if not G.is_directed() and weight is None:
         for v in nodes:
-            # Effective size is not defined for isolated nodes
-            if len(G[v]) == 0:
+            # Effective size is not defined for isolated nodes, including nodes
+            # with only self-edges
+            if (len(G[v]) == 0) or (set(G[v]) == {v}):
```

Comment:
How would you feel about this instead?
```suggestion
            if all(u == v for u in G[v]):
```
It's shorter, more readable (IMO) and short-circuits, which can be useful when there are lots of edges.

Reply:
Agreed - thanks for the suggestion!